File: problems/algorithms/0509/C/solution_formula.c
#include <math.h>

double PHI = 1.61803398875;
double SQRT_5 = 2.2360679775;

int fib(int n)
{
	if (n < 2)
		return n;
	return round((pow(PHI, n) - pow(1 - PHI, n)) / SQRT_5);
}

File: problems/algorithms/0509/Rust/solution_formula.rs
const PHI: f64 = 1.61803398875;
const SQRT_5: f64 = 2.2360679775;

impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n < 2 {
            return n;
        }
        ((PHI.powi(n) - (1.0 - PHI).powi(n))/SQRT_5).round() as i32
    }
}